packages/flutter_tools/lib/src/web/flutter_js.dart

```diff
@@ -82,6 +84,10 @@ _flutter.loader = null;
 
     _waitForServiceWorkerActivation(serviceWorker, entrypointUrl) {
       if (!serviceWorker) return;
```

Comment:
Is it ever safe to return here? That seems to guarantee a catastrophic result, since returning null results in a null engineInitializer.

Reply:
Maybe if we reach this case the safest thing to do is drop a plain `<script>` tag, same as the timeout fallback?

Reply:
> Maybe if we reach this case the safest thing to do is drop a plain <script> tag, same as the timeout fallback?

Yes, that sounds like a better fallback, let's add it.

Reply:
I slightly modified a second method to return null/a promise instead of being boolean, so we don't return "null" in one of the branches of the code.